provider/datagen/src/transform/segmenter/mod.rs

```diff
@@ -749,13 +749,27 @@ impl DataProvider<UCharDictionaryBreakDataV1Marker> for crate::DatagenProvider {
 
 impl IterableDataProvider<UCharDictionaryBreakDataV1Marker> for crate::DatagenProvider {
     fn supported_locales(&self) -> Result<Vec<DataLocale>, DataError> {
-        Ok(vec![
-            locale!("th").into(),
-            locale!("km").into(),
-            locale!("lo").into(),
-            locale!("my").into(),
-            locale!("ja").into(),
-        ])
+        let supported = [
+            langid!("th"),
+            langid!("km"),
+            langid!("lo"),
+            langid!("my"),
+            langid!("ja"),
+        ];
+
+        let needed = supported
+            .iter()
+            .filter(|l| self.source.options.dictionary_segmenter || langid!("ja") == **l);
+
+        use crate::options::LocaleInclude;
+        Ok(match &self.source.options.locales {
+            LocaleInclude::All => needed.map(DataLocale::from).collect(),
```

Comment:
Thought: I guess it makes sense for `LocaleInclude::All` to pull in all the dictionaries, too? But only because if you use that flag, you probably don't care about data size